lib/diagnostic.js

```diff
@@ -59,14 +61,20 @@ return array;
 // If the argument obj is unset, create a new object
 // The function you write should return
 // the object passed in as an argument or the object created
+function Obj(propertyName, method) {
+  this[propertyName] = method;
+}
 const addMethod = function addMethod(propertyName, method, obj) {
+
 if (obj) {
-  this.propertyName = method;
+  obj[propertyName] = method;
 } else {
-  obj = {
-    propertyName: method
-  };
+  let obj = new Obj(propertyName, method);
+  return obj;
+
 }
+
+return obj;
 };
```

Comment:
Your thinking is on the right track, but your syntax is off. Take a look at this solution:
```
const addMethod = function addMethod(propertyName, method, obj) {
  if (!obj) {
    obj = {};
  }
  obj[propertyName] = method;
  return obj;
};
```